pkg/validate/accounts.go

```diff
@@ -2,24 +2,40 @@ package validate
 
 import (
 	"context"
+	"fmt"
 	accounts "github.com/armory/spinnaker-operator/pkg/accounts"
 	"github.com/armory/spinnaker-operator/pkg/accounts/account"
 	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
 	"github.com/armory/spinnaker-operator/pkg/inspect"
+	"time"
 )
 
 // GetAccountValidationsFor inspects all known providers, retrieves their accounts,
 // and generate validators
 func GetAccountValidationsFor(spinSvc v1alpha2.SpinnakerServiceInterface, options Options) ([]SpinnakerValidator, error) {
 	validators := make([]SpinnakerValidator, 0)
 	for _, t := range accounts.Types {
+		v := t.GetValidationSettings(spinSvc)
+		if !v.Enabled {
+			continue
+		}
 		// Get accounts from that type
 		as, err := getAllAccounts(spinSvc, t, options)
 		if err != nil {
 			return nil, err
 		}
+		status := spinSvc.GetStatus()
+		now := time.Now()
 		for _, a := range as {
-			validators = append(validators, &accountValidator{v: a.NewValidator()})
+			h, err := a.GetHash()
+			if err != nil {
+				return nil, err
+			}
+			hc := status.UpdateHashIfNotExist(getValidationHashKey(a), h, now, false)
+			// If accounts were never validated or if the validation is too old less than x ago
+			if hc.Hash != h || !v.NeedsValidation(hc.LastUpdatedAt) {
```

Comment:
Shouldn't `!v.NeedsValidation(hc.LastUpdatedAt)` be NOT negated here?

Reply:
Oops thanks!